nntrainer/layers/pooling2d_layer.cpp

```diff
@@ -337,17 +359,47 @@ Tensor Pooling2DLayer::pooling2d(Tensor &in, bool training, Tensor &output) {
     };
     break;
   }
-  case PoolingType::average:
-  case PoolingType::global_max:
+  case PoolingType::global_max: {
+    break;
+  }
   case PoolingType::global_average:
+  case PoolingType::average: {
+    pool_fn = [&, this](const float *in_data, int start_h, int start_w) {
+      int end_h = start_h + patch_height;
+      int end_w = start_w + patch_width;
+      float total = 0.0f;
+      int cnt = 0;
+
+      for (int h = start_h; h < end_h; ++h) {
+        if (h < 0 || in_height <= h) {
+          continue;
+        }
+        for (int w = start_w; w < end_w; ++w) {
+          if (w < 0 || in_width <= w) {
+            continue;
+          }
+          float val = in_data[h * in_width + w];
+          cnt++;
```

Comment:
please calculate `cnt` statically than adding in every loop.

Reply:
Done